This fixes the issue for `cost` but not `nodecost`:

```
panic: reflect.Value.SetMapIndex: value of type int is not assignable to type float64
goroutine 1 [running]:
reflect.Value.assignTo(0x14b8a20, 0xc00029e600, 0x82, 0x1700408, 0x19, 0x1495ce0, 0x0, 0x14bbc20, 0xc0002d8f60, 0x98)
        /usr/local/go/src/reflect/value.go:2411 +0x426
reflect.Value.SetMapIndex(0x151c8a0, 0xc000398780, 0x15, 0x14bbc20, 0xc0002d8f60, 0x98, 0x14b8a20, 0xc00029e600, 0x82)
        /usr/local/go/src/reflect/value.go:1674 +0x1db
github.com/project-receptor/receptor/pkg/cmdline.setValue(0xc0001e3820, 0x151a2c0, 0xc000398600, 0xc0001e3820, 0x0)
        /home/squid/work/project-receptor/receptor/pkg/cmdline/cmdline.go:218 +0x885
github.com/project-receptor/receptor/pkg/cmdline.loadConfigFromFile(0x7ffcb07c15b6, 0x8, 0x7ffcb07c15b6, 0x8, 0xc0001d5dc8, 0x203000, 0x40deb6)
        /home/squid/work/project-receptor/receptor/pkg/cmdline/cmdline.go:448 +0x551
github.com/project-receptor/receptor/pkg/cmdline.ParseAndRun(0xc00003c190, 0x2, 0x2)
        /home/squid/work/project-receptor/receptor/pkg/cmdline/cmdline.go:517 +0x104f
main.main()
        /home/squid/work/project-receptor/receptor/cmd/receptor.go:60 +0x33c
```

--- a/pkg/cmdline/cmdline.go
+++ b/pkg/cmdline/cmdline.go
@@ -191,7 +191,7 @@ func setValue(field *reflect.Value, value interface{}) error {
 
 	// If the value is directly convertible to the field, just set it
 	if valueType.ConvertibleTo(fieldType) {
-		field.Set(reflect.ValueOf(value))
+		field.Set(reflect.ValueOf(value).Convert(fieldType))
 		return nil
 	}
 
@@ -211,25 +211,37 @@ func setValue(field *reflect.Value, value interface{}) error {
 	}
 
 	// If the field and value are a map type, attempt to copy the keys/values
-	if fieldKind == reflect.Map && valueType.Kind() == fieldKind {
+	if fieldKind == reflect.Map && valueType.Kind() == reflect.Map {
 		fieldMap := reflect.MakeMap(reflect.MapOf(fieldType.Key(), fieldType.Elem()))
 		iter := reflect.ValueOf(value).MapRange()
 		for iter.Next() {
-			fieldMap.SetMapIndex(reflect.ValueOf(iter.Key().Interface()), reflect.ValueOf(iter.Value().Interface()))
+			itemKey := reflect.ValueOf(iter.Key().Interface())
+			if !itemKey.Type().ConvertibleTo(fieldType.Key()) {
+				return fmt.Errorf("invalid key %s: must be type %s", itemKey, fieldType.Key())
+			}
+			itemValue := reflect.ValueOf(iter.Value().Interface())
+			if !itemValue.Type().ConvertibleTo(fieldType.Elem()) {
+				return fmt.Errorf("invalid value %s: must be type %s", itemValue, fieldType.Elem())
+			}
+			fieldMap.SetMapIndex(itemKey.Convert(fieldType.Key()), itemValue.Convert(fieldType.Elem()))
 		}
 		field.Set(fieldMap)
 		return nil
 	}
 
 	// If the field and value are a slice type, attempt to copy the values
-	if fieldKind == reflect.Slice && valueType.Kind() == fieldKind {
+	if fieldKind == reflect.Slice && valueType.Kind() == reflect.Slice {
 		valueSlice, ok := value.([]interface{})
 		if !ok {
 			return fmt.Errorf("invalid value for slice type")
 		}
 		fieldSlice := reflect.MakeSlice(fieldType, 0, 0)
 		for i := range valueSlice {
-			reflect.Append(fieldSlice, reflect.ValueOf(valueSlice[i]))
+			item := reflect.ValueOf(valueSlice[i])
+			if !item.Type().ConvertibleTo(fieldType.Elem()) {
+				return fmt.Errorf("invalid value %s: must be type %s", item, fieldType.Elem())
+			}
+			reflect.Append(fieldSlice, item.Convert(fieldType.Elem()))
 		}
 		field.Set(fieldSlice)
 		return nil
@@ -261,7 +273,7 @@ func setValue(field *reflect.Value, value interface{}) error {
 			return nil
 		}
 	}
-	return fmt.Errorf("type error")
+	return fmt.Errorf("type error (expected %s)", fieldType)
 }
 
 func plural(count int, singular string, plural string) string {

